### ISSUE TITLE
Migration sockets client from 0.9 to 1.2 with sails from 0.10 to 0.11

### ISSUE THREAD

**Issue by Vadorequest** [balderdashy/sails#3001]
I'm stuck at migrating my socket.io client.

### Sails 0.10
At the time, I used several files:

- socket.io.js https://gist.github.com/Vadorequest/54db7ad49c38ed0e80e7
- sails.io.js https://gist.github.com/Vadorequest/140d923891287700b3c5
- app.js https://gist.github.com/Vadorequest/a0fd02594ba7e52025d3

I was able to do stuff like `socket.emit`, `socket.get`...

### Sails 0.11
I now only have one file. Socket.io 1.2 is already included in sails.io.js

- sails.io.js https://gist.github.com/Vadorequest/be52414888db289271df
- *The app.js file shouldn't be used, right? Since it also opens a connection but `sails.io.js` already do.*

But it doesn't work anymore, there isn't any `window.socket` object, now the reference is in `window.io.socket`. But even using the right reference, I can't do any `io.socket.emit`, emit doesn't exist anymore.

And even a `io.socket.get` doesn't work, well, something is sent, the server receive the request somehow:

```
verbose: Receiving incoming message from Socket.io:  { method: 'get',
  headers: {},
  data: { sss: 5 },
  url: '/home/tellMyFriends' }
verbose: Interpreting socket.io message as virtual request to "get /home/tellMyFriends"...
```

But my controller Home.tellMyFriends isn't called at all, while it works through HTTP using the direct url.

So... How can I get the same behaviour as in sails 0.10?

**Comment by sailsbot** [comment]
Thanks for posting, @Vadorequest.  I'm a repo bot-- nice to meet you!

It has been 60 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!
